src/modules/map/components/openlayers/utils/print-area-renderer.composable.js

```diff
@@ -0,0 +1,207 @@
+import { Polygon } from 'ol/geom'
+import * as olHas from 'ol/has'
+import VectorLayer from 'ol/layer/Vector'
+import VectorSource from 'ol/source/Vector'
+import { computed, watch } from 'vue'
+import { useStore } from 'vuex'
+
+function createWorldPolygon() {
+    // Create a polygon feature covering the whole world in EPSG:4326
+    var worldPolygon = new Feature({
+        geometry: new Polygon([
+            [
+                [-180, -90], // Bottom-left corner
+                [180, -90], // Bottom-right corner
+                [180, 90], // Top-right corner
+                [-180, 90], // Top-left corner
+                [-180, -90], // Bottom-left corner
+            ],
+        ]).transform('EPSG:4326', 'EPSG:3857'),
+    })
+    // Create a VectorLayer outside the map creation
+    var vectorLayer = new VectorLayer({
+        source: new VectorSource({
+            features: [worldPolygon],
+        }),
+    })
+    return vectorLayer
+}
+
+export default function usePrintAreaRenderer(map) {
+    const store = useStore()
+
+    var deregister = []
+    var POINTS_PER_INCH = 72 // PostScript points 1/72"
+    var MM_PER_INCHES = 25.4
+    var UNITS_RATIO = 39.37 // inches per meter
+    var worldPolygon = null
+    var printRectangle = []
+
+    const isActive = computed(() => {
+        return store.state.print.printSectionShown
+    })
+
+    const selectedLayoutName = computed(() => {
+        return store.state.print.selectedLayout.name
+    })
+
+    const selectedScale = computed(() => {
+        return store.state.print.selectedScale
+    })
+
+    watch(isActive, (newValue) => {
+        if (newValue) {
+            activatePrintArea()
+        } else {
+            deactivatePrintArea()
+        }
+    })
+
+    function activatePrintArea() {
+        if (!worldPolygon) {
+            worldPolygon = createWorldPolygon()
+        }
+        map.addLayer(worldPolygon)
+        deregister = [
+            worldPolygon.on('prerender', handlePreRender),
+            worldPolygon.on('postrender', handlePostRender),
+            watch(selectedLayoutName, () => {
+                updatePrintRectanglePixels(selectedScale)
+            }),
+            watch(selectedScale, () => {
+                updatePrintRectanglePixels(selectedScale)
+            }),
+            map.on('change:size', () => {
+                updatePrintRectanglePixels(selectedScale)
+            }),
+            map.getView().on('propertychange', () => {
+                updatePrintRectanglePixels(selectedScale)
+            }),
+        ]
+        store.commit('setSelectedScale', getOptimalScale())
+        refreshComp()
+    }
+
+    function deactivatePrintArea() {
+        map.removeLayer(worldPolygon)
+        while (deregister.length > 0) {
+            var item = deregister.pop()
+            if (typeof item === 'function') {
+                item()
+            } else {
+                item.target.un(item.type, item.listener)
+            }
+        }
```

Comment:
I am not sure about using `.forEach` here since I also want to remove the element. Or do you mean empty the `deregister` (e.g. `deregister = []`) after `.forEach`?